starlite/testing.py

```diff
@@ -280,6 +281,188 @@ def test_my_handler() -> None:
     )
 
 
+class RequestFactory:
+    def __init__(
+        self,
+        app: Starlite = Starlite(route_handlers=[]),
+        server: str = "test.org",
+        port: int = 3000,
+        root_path: str = "",
+    ):
+        self.app = app
+        self.server = server
+        self.port = port
+        self.root_path = root_path
+
+    def _create_scope(
+        self,
+        path: str,
+        http_method: HttpMethod,
+        user: Any = None,
+        auth: Any = None,
+    ) -> Dict[str, Any]:
+        return dict(
+            type="http",
+            method=http_method,
+            server=(self.server, self.port),
+            root_path=self.root_path.rstrip("/"),
+            path=path,
+            headers=[],
+            app=self.app,
+            user=user,
+            auth=auth,
+        )
```

Comment:
Also add `session` here so that the user can set pre-loaded session for testing. The `session` here will be a dictionary that contains cookie key as keys and cookie values as its values. We need to check if user has passed cookies. If passed, make them into a dictionary and assign that value to `session` when returning scope here. See the comment below.